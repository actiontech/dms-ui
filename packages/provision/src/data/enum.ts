export enum ModalName {
  GetConnection = 'GET_CONNECTION',

  UpdateTemplateInAuth = 'UPDATE_TEMPLATE_IN_AUTH',

  UpdateUserInAuth = 'UPDATE_USER_IN_AUTH',

  UpdateExpirationInAuth = 'UPDATE_EXPIRATION_IN_AUTH',

  DataPermissionModal = 'DATA_PERMISSION_MODAL',

  CopyTemplate = 'COPY_TEMPLATE'
}

export enum EventEmitterKey {
  Refresh_Auth_List_Table = 'Refresh_Auth_List_Table',

  Refresh_Auth_Template_List_Table = 'Refresh_Auth_Template_List_Table'
}
