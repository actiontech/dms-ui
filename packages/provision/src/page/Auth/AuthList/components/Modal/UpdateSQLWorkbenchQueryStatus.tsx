import { useRecoilState } from 'recoil';
import {
  AuthListModalStatus,
  AuthListSelectData
} from '../../../../../store/auth/list';
import { useTranslation } from 'react-i18next';
import BasicDrawer from '@actiontech/shared/lib/components/BasicDrawer';
import { Form, Space, message } from 'antd';
import { BasicButton, BasicToolTips, BasicSwitch } from '@actiontech/shared';
import { EventEmitterKey, ModalName } from '../../../../../data/enum';
import useModalStatus from '../../../../../hooks/useModalStatus';
import { IUpdateSQLWorkbenchQueryFields } from '../List/index.type';
import { useBoolean } from 'ahooks';
import EventEmitter from '../../../../../utils/EventEmitter';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import auth from '@actiontech/shared/lib/api/provision/service/auth';
import { useEffect } from 'react';

const UpdateSQLWorkbenchQueryStatus: React.FC = () => {
  const { t } = useTranslation();
  const [messageApi, contextHolder] = message.useMessage();
  const { toggleModal, visible } = useModalStatus(
    AuthListModalStatus,
    ModalName.UpdateSQLWorkbenchQueryStatus
  );

  const [selectData, updateSelectData] = useRecoilState(AuthListSelectData);

  const [form] = Form.useForm<IUpdateSQLWorkbenchQueryFields>();

  const [updateLoading, { setTrue: startUpdate, setFalse: updateFinish }] =
    useBoolean();

  const closeModal = () => {
    toggleModal(ModalName.UpdateSQLWorkbenchQueryStatus, false);
    updateSelectData(null);
  };

  const closeAndReset = () => {
    closeModal();
    const timeId = setTimeout(() => {
      form.resetFields();
      clearTimeout(timeId);
    }, 300);
  };

  const submit = async () => {
    const values = await form.validateFields();
    startUpdate();
    try {
      const res = await auth.AuthUpdateAuthorization({
        authorization_uid: selectData?.uid ?? '',
        authorization: {
          used_by_sql_workbench: values.used_by_sql_workbench
        }
      });
      if (res.data.code === ResponseCode.SUCCESS) {
        messageApi.success(t('auth.updateSQLWorkbenchQueryStatus.successTips'));
        closeAndReset();
        EventEmitter.emit(EventEmitterKey.Refresh_Auth_List_Table);
      }
    } finally {
      updateFinish();
    }
  };

  useEffect(() => {
    if (visible) {
      form.setFieldValue(
        'used_by_sql_workbench',
        !!selectData?.used_by_sql_workbench
      );
    }
  }, [form, selectData?.used_by_sql_workbench, visible]);

  return (
    <BasicDrawer
      title={t('auth.updateSQLWorkbenchQueryStatus.title')}
      open={visible}
      footer={
        <Space>
          <BasicButton disabled={updateLoading} onClick={closeAndReset}>
            {t('common.close')}
          </BasicButton>
          <BasicButton type="primary" loading={updateLoading} onClick={submit}>
            {t('common.submit')}
          </BasicButton>
        </Space>
      }
      onClose={closeAndReset}
    >
      {contextHolder}
      <Form layout="vertical" form={form}>
        <Form.Item
          name="used_by_sql_workbench"
          valuePropName="checked"
          rules={[{ required: true }]}
          label={
            <BasicToolTips suffixIcon>
              {t('auth.updateSQLWorkbenchQueryStatus.label')}
            </BasicToolTips>
          }
        >
          <BasicSwitch />
        </Form.Item>
      </Form>
    </BasicDrawer>
  );
};

export default UpdateSQLWorkbenchQueryStatus;
