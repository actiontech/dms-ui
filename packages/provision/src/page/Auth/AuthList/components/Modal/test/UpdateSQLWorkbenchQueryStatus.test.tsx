import { superRender } from '@actiontech/shared/lib/testUtil/customRender';
import { act, cleanup, fireEvent, screen } from '@testing-library/react';
import { EventEmitterKey, ModalName } from '../../../../../../data/enum';
import {
  AuthListModalStatus,
  AuthListSelectData
} from '../../../../../../store/auth/list';
import { authorizationList } from '../../../../../../testUtil/mockApi/auth/data';
import UpdateSQLWorkbenchQueryStatus from '../UpdateSQLWorkbenchQueryStatus';
import auth from '../../../../../../testUtil/mockApi/auth';
import { getBySelector } from '../../../../../../testUtil/customQuery';
import eventEmitter from '../../../../../../utils/EventEmitter';

describe('page/Auth/AuthList/UpdateSQLWorkbenchQueryStatus', () => {
  const customRender = (
    defaultVisible = true,
    selectData = authorizationList[0]
  ) => {
    return superRender(<UpdateSQLWorkbenchQueryStatus />, undefined, {
      recoilRootProps: {
        initializeState({ set }) {
          set(AuthListModalStatus, {
            [ModalName.UpdateSQLWorkbenchQueryStatus]: defaultVisible
          });
          set(AuthListSelectData, selectData);
        }
      }
    });
  };

  beforeEach(() => {
    jest.useFakeTimers();
    auth.mockAllApi();
  });

  afterEach(() => {
    jest.useRealTimers();
    cleanup();
  });

  it('render init snapshot', () => {
    const { baseElement } = customRender(true);

    expect(baseElement).toMatchSnapshot();
  });

  it('render init snapshot when "used_by_sql_workbench" is equal true', () => {
    const { baseElement } = customRender(true, authorizationList[2]);

    expect(baseElement).toMatchSnapshot();
  });

  it('render close icon btn', async () => {
    const { baseElement } = customRender();

    const closeIcon = getBySelector('.anticon-close', baseElement);
    fireEvent.click(closeIcon);
    await act(async () => jest.advanceTimersByTime(0));
    expect(baseElement).toMatchSnapshot();
  });

  it('render close text btn', async () => {
    const { baseElement } = customRender();
    fireEvent.click(screen.getByText('关 闭'));
    await act(() => jest.advanceTimersByTime(0));
    expect(baseElement).toMatchSnapshot();
  });

  it('render submit', async () => {
    const emitSpy = jest.spyOn(eventEmitter, 'emit');
    const request = auth.updateAuthorization();

    const { baseElement } = customRender(true);

    fireEvent.click(getBySelector('#used_by_sql_workbench'));

    fireEvent.click(screen.getByText('提 交'));
    await act(() => jest.advanceTimersByTime(0));

    expect(request).toHaveBeenCalledTimes(1);
    expect(request).toHaveBeenCalledWith({
      authorization_uid: authorizationList[0]?.uid ?? '',
      authorization: {
        used_by_sql_workbench: true
      }
    });
    await act(async () => {
      jest.advanceTimersByTime(3000);
      await act(() => jest.advanceTimersByTime(300));
    });

    expect(screen.getByText('更新成功')).toBeInTheDocument();
    expect(emitSpy).toHaveBeenCalledTimes(1);
    expect(emitSpy).toHaveBeenCalledWith(
      EventEmitterKey.Refresh_Auth_List_Table
    );
    await act(() => jest.advanceTimersByTime(0));
    expect(baseElement).toMatchSnapshot();
  });
});
