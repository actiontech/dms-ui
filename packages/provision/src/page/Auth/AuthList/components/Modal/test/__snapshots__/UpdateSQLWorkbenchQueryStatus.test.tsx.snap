// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page/Auth/AuthList/UpdateSQLWorkbenchQueryStatus render close icon btn 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-drawer ant-drawer-right css-dev-only-do-not-override-txh9fw"
      tabindex="-1"
    >
      <div
        aria-hidden="true"
        data-sentinel="start"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
      <div
        class="ant-drawer-content-wrapper ant-drawer-content-wrapper-hidden"
        style="width: 480px;"
      >
        <div
          aria-modal="true"
          class="ant-drawer-content basic-drawer-wrapper css-1q29jje"
          role="dialog"
        >
          <div
            class="ant-drawer-wrapper-body"
          >
            <div
              class="ant-drawer-header"
            >
              <div
                class="ant-drawer-header-title"
              >
                <div
                  class="ant-drawer-title"
                >
                  变更SQL工作台查询起停状态
                </div>
              </div>
            </div>
            <div
              class="ant-drawer-body"
            >
              <div
                class="closed-icon-custom"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                    />
                  </svg>
                </span>
              </div>
              <form
                class="ant-form ant-form-vertical css-dev-only-do-not-override-txh9fw"
              >
                <div
                  class="ant-form-item css-dev-only-do-not-override-txh9fw"
                >
                  <div
                    class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
                  >
                    <div
                      class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
                    >
                      <label
                        class="ant-form-item-required"
                        for="used_by_sql_workbench"
                        title=""
                      >
                        是否用于SQL工作台查询
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            aria-checked="false"
                            aria-required="true"
                            class="ant-switch basic-switch-wrapper css-g6dhbn css-dev-only-do-not-override-txh9fw"
                            id="used_by_sql_workbench"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            >
                              <span
                                class="ant-switch-inner-checked"
                              />
                              <span
                                class="ant-switch-inner-unchecked"
                              />
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-drawer-footer"
            >
              <div
                class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      关 闭
                    </span>
                  </button>
                </div>
                <div
                  class="ant-space-item"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      提 交
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-sentinel="end"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;

exports[`page/Auth/AuthList/UpdateSQLWorkbenchQueryStatus render close text btn 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-drawer ant-drawer-right css-dev-only-do-not-override-txh9fw"
      tabindex="-1"
    >
      <div
        aria-hidden="true"
        data-sentinel="start"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
      <div
        class="ant-drawer-content-wrapper ant-drawer-content-wrapper-hidden"
        style="width: 480px;"
      >
        <div
          aria-modal="true"
          class="ant-drawer-content basic-drawer-wrapper css-1q29jje"
          role="dialog"
        >
          <div
            class="ant-drawer-wrapper-body"
          >
            <div
              class="ant-drawer-header"
            >
              <div
                class="ant-drawer-header-title"
              >
                <div
                  class="ant-drawer-title"
                >
                  变更SQL工作台查询起停状态
                </div>
              </div>
            </div>
            <div
              class="ant-drawer-body"
            >
              <div
                class="closed-icon-custom"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                    />
                  </svg>
                </span>
              </div>
              <form
                class="ant-form ant-form-vertical css-dev-only-do-not-override-txh9fw"
              >
                <div
                  class="ant-form-item css-dev-only-do-not-override-txh9fw"
                >
                  <div
                    class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
                  >
                    <div
                      class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
                    >
                      <label
                        class="ant-form-item-required"
                        for="used_by_sql_workbench"
                        title=""
                      >
                        是否用于SQL工作台查询
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            aria-checked="false"
                            aria-required="true"
                            class="ant-switch basic-switch-wrapper css-g6dhbn css-dev-only-do-not-override-txh9fw"
                            id="used_by_sql_workbench"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            >
                              <span
                                class="ant-switch-inner-checked"
                              />
                              <span
                                class="ant-switch-inner-unchecked"
                              />
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-drawer-footer"
            >
              <div
                class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      关 闭
                    </span>
                  </button>
                </div>
                <div
                  class="ant-space-item"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      提 交
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-sentinel="end"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;

exports[`page/Auth/AuthList/UpdateSQLWorkbenchQueryStatus render init snapshot 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-drawer ant-drawer-right css-dev-only-do-not-override-txh9fw ant-drawer-open"
      tabindex="-1"
    >
      <div
        class="ant-drawer-mask"
      />
      <div
        aria-hidden="true"
        data-sentinel="start"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
      <div
        class="ant-drawer-content-wrapper"
        style="width: 480px;"
      >
        <div
          aria-modal="true"
          class="ant-drawer-content basic-drawer-wrapper css-1q29jje"
          role="dialog"
        >
          <div
            class="ant-drawer-wrapper-body"
          >
            <div
              class="ant-drawer-header"
            >
              <div
                class="ant-drawer-header-title"
              >
                <div
                  class="ant-drawer-title"
                >
                  变更SQL工作台查询起停状态
                </div>
              </div>
            </div>
            <div
              class="ant-drawer-body"
            >
              <div
                class="closed-icon-custom"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                    />
                  </svg>
                </span>
              </div>
              <form
                class="ant-form ant-form-vertical css-dev-only-do-not-override-txh9fw"
              >
                <div
                  class="ant-form-item css-dev-only-do-not-override-txh9fw"
                >
                  <div
                    class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
                  >
                    <div
                      class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
                    >
                      <label
                        class="ant-form-item-required"
                        for="used_by_sql_workbench"
                        title=""
                      >
                        是否用于SQL工作台查询
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            aria-checked="false"
                            aria-required="true"
                            class="ant-switch basic-switch-wrapper css-g6dhbn css-dev-only-do-not-override-txh9fw"
                            id="used_by_sql_workbench"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            >
                              <span
                                class="ant-switch-inner-checked"
                              />
                              <span
                                class="ant-switch-inner-unchecked"
                              />
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-drawer-footer"
            >
              <div
                class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      关 闭
                    </span>
                  </button>
                </div>
                <div
                  class="ant-space-item"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      提 交
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-sentinel="end"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;

exports[`page/Auth/AuthList/UpdateSQLWorkbenchQueryStatus render init snapshot when "used_by_sql_workbench" is equal true 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-drawer ant-drawer-right css-dev-only-do-not-override-txh9fw ant-drawer-open"
      tabindex="-1"
    >
      <div
        class="ant-drawer-mask"
      />
      <div
        aria-hidden="true"
        data-sentinel="start"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
      <div
        class="ant-drawer-content-wrapper"
        style="width: 480px;"
      >
        <div
          aria-modal="true"
          class="ant-drawer-content basic-drawer-wrapper css-1q29jje"
          role="dialog"
        >
          <div
            class="ant-drawer-wrapper-body"
          >
            <div
              class="ant-drawer-header"
            >
              <div
                class="ant-drawer-header-title"
              >
                <div
                  class="ant-drawer-title"
                >
                  变更SQL工作台查询起停状态
                </div>
              </div>
            </div>
            <div
              class="ant-drawer-body"
            >
              <div
                class="closed-icon-custom"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                    />
                  </svg>
                </span>
              </div>
              <form
                class="ant-form ant-form-vertical css-dev-only-do-not-override-txh9fw"
              >
                <div
                  class="ant-form-item css-dev-only-do-not-override-txh9fw"
                >
                  <div
                    class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
                  >
                    <div
                      class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
                    >
                      <label
                        class="ant-form-item-required"
                        for="used_by_sql_workbench"
                        title=""
                      >
                        是否用于SQL工作台查询
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            aria-checked="false"
                            aria-required="true"
                            class="ant-switch basic-switch-wrapper css-g6dhbn css-dev-only-do-not-override-txh9fw"
                            id="used_by_sql_workbench"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            >
                              <span
                                class="ant-switch-inner-checked"
                              />
                              <span
                                class="ant-switch-inner-unchecked"
                              />
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-drawer-footer"
            >
              <div
                class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      关 闭
                    </span>
                  </button>
                </div>
                <div
                  class="ant-space-item"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      提 交
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-sentinel="end"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;

exports[`page/Auth/AuthList/UpdateSQLWorkbenchQueryStatus render submit 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-drawer ant-drawer-right css-dev-only-do-not-override-txh9fw"
      tabindex="-1"
    >
      <div
        aria-hidden="true"
        data-sentinel="start"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
      <div
        class="ant-drawer-content-wrapper ant-drawer-content-wrapper-hidden"
        style="width: 480px;"
      >
        <div
          aria-modal="true"
          class="ant-drawer-content basic-drawer-wrapper css-1q29jje"
          role="dialog"
        >
          <div
            class="ant-drawer-wrapper-body"
          >
            <div
              class="ant-drawer-header"
            >
              <div
                class="ant-drawer-header-title"
              >
                <div
                  class="ant-drawer-title"
                >
                  变更SQL工作台查询起停状态
                </div>
              </div>
            </div>
            <div
              class="ant-drawer-body"
            >
              <div
                class="closed-icon-custom"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                    />
                  </svg>
                </span>
              </div>
              <form
                class="ant-form ant-form-vertical css-dev-only-do-not-override-txh9fw"
              >
                <div
                  class="ant-form-item css-dev-only-do-not-override-txh9fw"
                >
                  <div
                    class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
                  >
                    <div
                      class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
                    >
                      <label
                        class="ant-form-item-required"
                        for="used_by_sql_workbench"
                        title=""
                      >
                        是否用于SQL工作台查询
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            aria-checked="false"
                            aria-required="true"
                            class="ant-switch basic-switch-wrapper css-g6dhbn css-dev-only-do-not-override-txh9fw"
                            id="used_by_sql_workbench"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            >
                              <span
                                class="ant-switch-inner-checked"
                              />
                              <span
                                class="ant-switch-inner-unchecked"
                              />
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-drawer-footer"
            >
              <div
                class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      关 闭
                    </span>
                  </button>
                </div>
                <div
                  class="ant-space-item"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
                    type="button"
                  >
                    <span>
                      提 交
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-sentinel="end"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
    </div>
  </div>
  <div
    class="ant-message ant-message-top css-dev-only-do-not-override-txh9fw"
    style="left: 50%; transform: translateX(-50%); top: 8px;"
  >
    <div
      class="ant-message-notice ant-message-notice-success css-dev-only-do-not-override-txh9fw"
    >
      <div
        class="ant-message-notice-content"
      >
        <div
          class="ant-message-custom-content ant-message-success"
        >
          <span
            aria-label="check-circle"
            class="anticon anticon-check-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="check-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
              />
            </svg>
          </span>
          <span>
            更新成功
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
